src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificatePolicy.cs

```diff
@@ -0,0 +1,378 @@
+            ISet<string> ekus = null;
+            CertificatePolicy policy = new CertificatePolicy();
+
+            foreach (X509Extension extension in cert.Extensions)
+            {
+                switch (extension.Oid.Value)
+                {
+                    case Oids.ApplicationCertPolicies:
+                        applicationCertPolicies = ReadCertPolicyExtension(extension);
+                        break;
+                    case Oids.CertPolicies:
+                        policy.DeclaredCertificatePolicies = ReadCertPolicyExtension(extension);
+                        break;
+                    case Oids.CertPolicyMappings:
+                        policy.PolicyMapping = ReadCertPolicyMappingsExtension(extension);
+                        break;
+                    case Oids.CertPolicyConstraints:
+                        ReadCertPolicyConstraingsExtension(extension, policy);
+                        break;
+                    case Oids.EnhancedKeyUsage:
+                        if (applicationCertPolicies == null)
+                        {
+                            // No reason to do this if the applicationCertPolicies was already read
+                            ekus = ReadExtendedKeyUsageExtension(extension);
+                        }
+
+                        break;
+                    case Oids.InhibitAnyPolicyExtension:
+                        policy.InhibitAnyDepth = ReadInhibitAnyPolicyExtension(extension);
+                        break;
+                }
+            }
+
+            policy.DeclaredApplicationPolicies = applicationCertPolicies ?? ekus;
+
+            policy.ImplicitAnyApplicationPolicy = policy.DeclaredApplicationPolicies == null;
+            policy.ImplicitAnyCertificatePolicy = policy.DeclaredCertificatePolicies == null;
+
+            policy.SpecifiedAnyApplicationPolicy = CheckExplicitAnyPolicy(policy.DeclaredApplicationPolicies);
+            policy.SpecifiedAnyCertificatePolicy = CheckExplicitAnyPolicy(policy.DeclaredCertificatePolicies);
+
+            return policy;
+        }
+
+        private static bool CheckExplicitAnyPolicy(ISet<string> declaredPolicies)
+        {
+            if (declaredPolicies == null)
+            {
+                return false;
+            }
+
+            return declaredPolicies.Remove(Oids.AnyCertPolicy);
+        }
+
+        private static int ReadInhibitAnyPolicyExtension(X509Extension extension)
+        {
+            DerSequenceReader reader = DerSequenceReader.CreateForPayload(extension.RawData);
+            return reader.ReadInteger();
+        }
+
+        private static void ReadCertPolicyConstraingsExtension(X509Extension extension, CertificatePolicy policy)
+        {
+            DerSequenceReader reader = new DerSequenceReader(extension.RawData);
+
+            while (reader.HasData)
+            {
+                switch (reader.PeekTag())
+                {
+                    case 0x80:
```

Comment:
This number should be explained. Something like:
`Identifier of context-specific, primitive tag for field 0 in PolicyConstraints (see X.509 section 4.2.1.11)`